File: src/main/java/com/dh/ondot/schedule/domain/enums/Mission.java
package com.dh.ondot.schedule.domain.enums;

import com.dh.ondot.core.exception.ErrorCode;
import com.dh.ondot.core.exception.UnsupportedException;

import java.util.Locale;

public enum Mission {
    NONE;

    public static Mission from(String mission) {
        try {
            return Mission.valueOf(mission.toUpperCase(Locale.ENGLISH));
        } catch (IllegalArgumentException e) {
            throw new UnsupportedException(ErrorCode.UNSUPPORTED_MISSION, mission);
        }
    }
}

File: src/main/kotlin/com/dh/ondot/schedule/domain/enums/Mission.kt
package com.dh.ondot.schedule.domain.enums

import com.dh.ondot.core.exception.ErrorCode
import com.dh.ondot.core.exception.UnsupportedException
import java.util.Locale

enum class Mission {
    NONE;

    companion object {
        @JvmStatic
        fun from(mission: String): Mission =
            try {
                valueOf(mission.uppercase(Locale.ENGLISH))
            } catch (e: IllegalArgumentException) {
                throw UnsupportedException(ErrorCode.UNSUPPORTED_MISSION, mission)
            }
    }
}
